Marlin/src/module/tool_change.cpp

```diff
@@ -443,12 +443,8 @@ inline void invalid_extruder_error(const uint8_t e) {
         inactive_extruder_x_pos = destination[X_AXIS];
         break;
       case DXC_AUTO_PARK_MODE:
-        // record raised toolhead position for use by unpark
+        // record current raised toolhead position for use by unpark
         COPY(raised_parked_position, current_position);
-        raised_parked_position[Z_AXIS] += TOOLCHANGE_UNPARK_ZLIFT;
-        #if ENABLED(MAX_SOFTWARE_ENDSTOPS)
-          NOMORE(raised_parked_position[Z_AXIS], soft_endstop_max[Z_AXIS]);
-        #endif
```

Comment:
This raise move is being applied in the conditional above covered by a limit statement already. Since the z lower is not being performed between tools, the 2nd z raise is not necessary any longer. Were taking the current position as is, and no longer modifying it so the check is redundant.